packages/vetrina/src/Vetrina/Purchase/NewPurchase.purs

```diff
@@ -36,6 +37,7 @@ type State =
        }
   , accountStatus       :: AccountStatus
   , serverErrors        :: Array (Form.ValidationError FormInputField)
+  , displayErrorMessage :: Boolean
```

Comment:
We kind of have the "to show or not to show the error message" logic already in the `errorMessage :: Maybe String` type we get from props. So instead of having this boolean, we could just have `errorMessage :: Maybe String` (or perhaps even `Maybe JSX`) in the state, and change the calls to `self.props.errorMessage` to `self.state.errorMessage`.

Reply:
To this argument I counter that we already have the the message in the props, but either is fine for me

Reply:
And thinking more about it, the `errorMessage :: Maybe String` in the only tells us whether, when we instantiate the component from the `vetrina` parent component, we need to display the message. But if we need to hide it later is not determined by it.

But I'm doing as you say and this does simplify the code

Reply:
I think it's more descriptive to deal straight with Maybes if possible, and avoid adding Booleans to the mix. In my opinion the Boolean does not bring much to the table here. If we want to make changes to the `errorMessage` it's perfectly fine to have it in the state.

If we use Maybe, it's pretty straight forward as we have now:
```purescript
if (isJust errorMessage)
then renderErrorMessage errorMessage
else mempty
```
But for instance, if we rely on a combination of Maybe and Boolean, we can end up in a situation of `errorMessage = Nothing` and `displayErrorMessage = true` which would not make sense.

Reply:
> the props only tells us whether, when we instantiate the component from the vetrina parent component, we need to display the message. But if we need to hide it later is not determined by it.

Exactly